disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -35,6 +39,13 @@ maybeGen g = sized $ \s ->
     (1, return Nothing),
     (s, Just <$> resize (s `div` 2) g)]
 
+-- | Wait for a generated `Just` value
+--
+-- Use _only_ in case of emergencies when you have no other way to get an `a` safely
+genFromMaybe :: Gen (Maybe a) -> Gen a
```

Comment:
If you are going to do this shouldn't it fail the generator rather than throw isNothing exception. Think this would be quite puzzling to debug.